### PR TITLE
Prevent *.twig, config.xml & config_{lang}.xml files from direct access in modules folder

### LINKED ISSUE #29422: Do not authorize config.xml direct call on modules directory
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

If you call a shop with URL https://www.yourshop.com/modules/module_name/config.xml (or config_en.xml) you can know with version of the module is installed.

For instance :
- https://www.yourshop.com/modules/blockwishlist/config.xml

In case of security advisory on a module, a hacker can know if the PrestaShop has a module not up-to-date.

### Expected behavior

Return a forbidden access to this URL with the htaccess.

### Steps to reproduce

1. Install a new PrestaShop
2. Install the module blockwishlist (or an other)
3. Go to URL like https://www.yourshop.com/modules/blockwishlist/config.xml

### PrestaShop version(s) where the bug happened

all

### PHP version(s) where the bug happened

all

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by clotairer**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | If you call a shop with URL https://www.yourshop.com/modules/module_name/config.xml (or config_en.xml) you can know with version of the module is installed.
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #29422
| Related PRs       | nc
| How to test?      | * Install a new PrestaShop<br>* Install the module blockwishlist (or an other)<br>* Go to URL like https://www.yourshop.com/modules/blockwishlist/config.xml
| Possible impacts? | None

### CODE REVIEW

File: modules/.htaccess [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<FilesMatch "\.tpl$">
+<FilesMatch "\.tpl|\.twig|config\.xml|config_([a-z]+)\.xml$">
```
Comment (atomiix) [inline modules/.htaccess]: If you don't put the `()` only `config_([a-z]+)\.xml` will match the end of line ;)
```suggestion
<FilesMatch "(\.tpl|\.twig|config\.xml|config_([a-z]+)\.xml)$">
```

**Review by atomiix — APPROVED** [review]

**Comment by clotairer** [comment]
Thank you for your help @atomiix !

**Review by matks — APPROVED** [review]
Thank you @clotairer this is great

**Review by davidPrestashop — APPROVED** [review]
Tested on the blockwishlist module 
Result on the develop branch:
Screenshot from 2022-10-27 15-52-51

PR result:
Screenshot from 2022-10-27 15-52-33

**Pull request merged**